x-pack/plugins/observability/public/hooks/use_summary_time_range.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { TimeRange } from '@kbn/es-query';
+import { useUiSetting } from '@kbn/kibana-react-plugin/public';
+import { calculateTimeRangeBucketSize, getAlertSummaryTimeRange, useTimeBuckets } from '..';
+import { DEFAULT_INTERVAL, DEFAULT_DATE_FORMAT } from '../constants';
+
+export const useSummaryTimeRange = (unifiedSearchDateRange: TimeRange) => {
```

Comment:
Can we use this abstraction in other usages of AlertSummaryWidget? It would be nice to improve this part also for other AlertSummaryWidgets if it is possible :)

Reply:
If you think it can help on the alerts and overview page feel free to open an issue to replace it there after this PR is merged 🙂 I am not super familiar with the implementation there so I would say that that's a bit out of the scope of this PR (already getting big and I see it's hard to get a review already 😅 )